site/src/main/java/uk/nhs/digital/ps/beans/Dataset.java

```diff
@@ -95,4 +109,42 @@ public Publication getParentPublication() {
 
         return publicationBean;
     }
+
+    private <T> T getPropertyIfPermitted(final String propertyKey) {
+        assertPropertyPermitted(propertyKey);
+
+        return getProperty(propertyKey);
+    }
+
+    private void assertPropertyPermitted(final String propertyKey) {
+
+        final boolean isPropertyPermitted =
+            isParentPublicationPubliclyAccessible()
+                || propertiesPermittedWhenUpcoming.contains(propertyKey);
+
+        if (!isPropertyPermitted) {
+            throw new DataRestrictionViolationException(
+                "Property is not available when parent publication is flagged as 'not publicly accessible': " + propertyKey
+            );
+        }
+    }
+
+    public boolean isParentPublicationPubliclyAccessible() {
```

Comment:
I wouldn't reveal implementation detail in the function name, `isPubliclyAccessible()` is absolutely enough in my opinion.

Reply:
Fair point, will amend.